### PR TITLE
mnist dataset falls back to its original website

### PR DESCRIPTION
**Pull request by facaiy**
I find sometimes the network connection becomes unstable when visiting storage.googleapis.com in China, and it results in download error. That's why I want to make it fall back to the original website.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,4 +17,5 @@ dist/
 .pytest_cache/
 
 # Other
-*.DS_Store
\ No newline at end of file
+*.DS_Store
```
Comment (us) [inline .gitignore]: Please that .gitignore changes.

File: tensorflow_datasets/image/mnist.py [hunk 2]
```diff
@@ -111,21 +112,27 @@ def _info(self):
             "label": tfds.features.ClassLabel(num_classes=10),
         }),
         supervised_keys=("image", "label"),
-        urls=[self.URL],
+        urls=[self.URL, _MNIST_URL_FALLBACK_],
```
Comment (us) [inline tensorflow_datasets/image/mnist.py]: It's a main page of mnist not a download link.

File: tensorflow_datasets/image/mnist.py [hunk 3]
```diff
@@ -111,21 +112,27 @@ def _info(self):
...
     # Download the full MNIST Database
-    filenames = {
-        "train_data": _MNIST_TRAIN_DATA_FILENAME,
-        "train_labels": _MNIST_TRAIN_LABELS_FILENAME,
-        "test_data": _MNIST_TEST_DATA_FILENAME,
-        "test_labels": _MNIST_TEST_LABELS_FILENAME,
-    }
-    mnist_files = dl_manager.download_and_extract(
-        {k: urllib.parse.urljoin(self.URL, v) for k, v in filenames.items()})
+    def download_and_extract(url):
```
Comment (us) [inline tensorflow_datasets/image/mnist.py]: Why create a new function?

**Comment by us** [comment]
Thank you for contributing.

File: .gitignore (hunk 1 above)
Comment (facaiy) [reply .gitignore]: bypass pycharm(popular Python IDE) configurations.

File: tensorflow_datasets/image/mnist.py (hunk 3 above)
Comment (facaiy) [reply tensorflow_datasets/image/mnist.py]: I think it makes try...except block easy to understand. So, let's remove the function or not?

**Comment by facaiy** [comment]
@us thank you, can you take a look again?

File: tensorflow_datasets/image/mnist.py (hunk 3 above)
Comment (us) [reply tensorflow_datasets/image/mnist.py]: I think inner function is confusing so you can delete. 

You can split `dl_manager.download` and `dl_manager.extract` functions. Now try..except block is be better.

File: .gitignore (hunk 1 above)
Comment (facaiy) [reply .gitignore]: done, please to check #739

File: tensorflow_datasets/image/mnist.py (hunk 3 above)
Comment (facaiy) [reply tensorflow_datasets/image/mnist.py]: done :-)

File: tensorflow_datasets/image/mnist.py [hunk 4]
```diff
@@ -124,8 +125,15 @@ def _split_generators(self, dl_manager):
         "test_data": _MNIST_TEST_DATA_FILENAME,
         "test_labels": _MNIST_TEST_LABELS_FILENAME,
     }
-    mnist_files = dl_manager.download_and_extract(
-        {k: urllib.parse.urljoin(self.URL, v) for k, v in filenames.items()})
+
+    try:
```
Comment (us) [inline tensorflow_datasets/image/mnist.py]: the urls data have same folder structure so we can use try..exception block for just a `dl_manager.download()`, then use the `dl_manager.extract` method. 

And you need to be more clear when catch exception, Exception is too general for dl_manager. You should just catch the `tfds.core.download.downloader.DownloadError`.

Comment (facaiy) [reply tensorflow_datasets/image/mnist.py]: Thank you, fix it

**Pull request closed without merging**